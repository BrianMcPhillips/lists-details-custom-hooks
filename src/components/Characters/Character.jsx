import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ 
  name,
  status,
  occupation,
  image,
  description
}) => (
  <>
    <h2>{name}</h2>
    <p>{status}</p>
    <p>{occupation}</p>
    <img src={image} alt={name}/>
    <p>{description}</p>
  </>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  occupation: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Character;
