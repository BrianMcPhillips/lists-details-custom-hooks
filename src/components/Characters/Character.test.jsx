import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders character component', () => {
    const { asFragment } = render(<Character 
      name="Test Testerson"
      status="Meh"
      occupation="Crushing it full time"
      image="http://www.placecage.com/200/200"
      description="this is the best test ever written"
    />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
