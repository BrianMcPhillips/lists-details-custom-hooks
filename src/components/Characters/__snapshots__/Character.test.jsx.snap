// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders character component 1`] = `
<DocumentFragment>
  <h2>
    Test Testerson
  </h2>
  <img
    alt="Test Testerson"
    src="http://www.placecage.com/200/200"
  />
</DocumentFragment>
`;
